File: Exemplos/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/SAT/AtivarSAT.ts
import {SatCommand} from './SatCommand';

export class AtivarSAT extends SatCommand {
  readonly numSessao: number;
  readonly subComando: number;
  readonly codAtivacao: string;
  readonly cnpj: string;
  readonly cUF: number;

  constructor(
    numSessao: number,
    subComando: number,
    codAtivacao: string,
    cnpj: string,
    cUF: number,
  ) {
    super('AtivarSAT');
    this.numSessao = numSessao;
    this.subComando = subComando;
    this.codAtivacao = codAtivacao;
    this.cnpj = cnpj;
    this.cUF = cUF;
  }

  functionParameters(): object {
    return {
      numSessao: this.numSessao,
      subComando: this.subComando,
      codAtivacao: this.codAtivacao,
      cnpj: this.cnpj,
      cUF: this.cUF,
    };
  }
}

File: Exemplos M10/App_IntentDigitalHUB_XamarinAndroid/Elgin_IntentDigitalHUB_XamarinAndroid/IntentServices/Sat/AtivarSAT.cs
namespace Xamarin_Android_Intent_Digital_Hub.IntentServices.Sat
{
    class AtivarSAT : SatCommand
    {
        readonly int numSessao;
        readonly int subComando;
        readonly string codAtivacao;
        readonly string cnpj;
        readonly int cUF;

        public AtivarSAT(int numSessao, int subComando, string codAtivacao, string cnpj, int cUF) : base("AtivarSAT")
        {
            this.numSessao = numSessao;
            this.subComando = subComando;
            this.codAtivacao = codAtivacao;
            this.cnpj = cnpj;
            this.cUF = cUF;
        }

        protected override string FunctionParameters()
        {
            return "\"numSessao\"" + ":" + numSessao + "," +
                    "\"subComando\"" + ":" + subComando + "," +
                    "\"codAtivacao\"" + ":" + "\"" + codAtivacao + "\"" + "," +
                    "\"cnpj\"" + ":" + "\"" + cnpj + "\"" + "," +
                    "\"cUF\"" + ":" + cUF;
        }
    }
}